media/src/App.svelte

```diff
@@ -448,17 +447,18 @@
 
                 chatBuffer = [...chatBuffer, data.message.chatMessage];
             }
-        } else if (
-            data.message.pollResponse &&
-            !(data.session_key in userResponseList)
-        ) {
-            userResponseList[data.session_key] = true;
-            let choice = data.message.pollResponse;
-            if (choice in pollResponses) {
-                pollResponses[choice]++;
-            } else {
-                pollResponses[choice] = 1;
+        } else if (data.message.pollResponse) {
+            const sessionKey = data.message.session_key;
+            let responseObj = {
+                choice: data.message.pollResponse,
+                session_key: sessionKey
+            };
+            // Remove any previous response from this session.
+            if (pollResponses.find(x => x.session_key === sessionKey)) {
+                pollResponses = pollResponses.filter(
+                    x => x.session_key !== sessionKey);
```

Comment:
If you're doing a lookup using the session key, why not keep it as an object with `session_key` as the key and one of the object properties? It should be a faster lookup and you wouldn't have to delete the object in order to overwrite the response.

Regardless you don't need to filter out the object from the array either you can overwrite `x.choice`.

Reply:
Arrays are just nicer to deal with than objects and it's easier to process the data to get into d3. In this scenario I'm focusing on making the code simple and easier to understand - any performance optimizations can be added later if necessary, but it's not really an issue at this scale.

Reply:
But yeah - I did think of this alternative structure here.. hmm:

```
let pollResponses = {
  'session_key_1': 'a',
  'session_key_2': 'b'
};
```

Reply:
I'll do some adjustments here and see how that feels and get back to this. Converting this to a draft for now.